packages/core/src/tabs/tab-group.ts

```diff
@@ -0,0 +1,248 @@
+import type { CSSResultGroup, TemplateResult } from 'lit';
+import type { EventEmitter } from '../utils/decorators/event.js';
+import type { ScopedElementsMap } from '@open-wc/scoped-elements';
+import { ScopedElementsMixin } from '@open-wc/scoped-elements';
+import { LitElement, html } from 'lit';
+import { property, queryAssignedElements, state } from 'lit/decorators.js';
+import { event } from '../utils/decorators/event.js';
+import { RovingTabindexController } from '../utils/controllers/roving-tabindex.js';
+import { Tab } from './tab.js';
+import styles from './tab-group.scss.js';
+import { TabPanel } from './tab-panel.js';
+
+let tabGroupCount = 0;
+
+export class TabGroup extends ScopedElementsMixin(LitElement) {
+  /** @private */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-tab': Tab,
+      'sl-tab-panel': TabPanel
+    };
+  }
+
+  /** @private */
+  static override styles: CSSResultGroup = styles;
+
+  /**
+   * Unique ID for each tab group component present.
+   */
+  private tabGroupId = `sl-tab-group-${tabGroupCount++}`;
+
+  private observer?: MutationObserver;
```

Comment:
Please use `#observer` instead of the typescript `private` keyword. The first is pure JS, the latter is TS-only :)